### DISCUSSION TITLE
Is there a way to do a dryrun and print out screenshot count ?

### DISCUSSION THREAD

**Discussion by christoffee** [percy/percy-storybook#617] [Support] [answered]
Hey,

We have integrated percy into our component library... and we are producing a lot of screenshots.

Is there a way when running dryrun it prints out the screenshot count ?

**Comment by wwilsman** [comment, marked as answer]
Hey @christoffee!

With the `--dry-run` flag, it will print out the snapshot count at the end:

<img width="201" alt="CleanShot 2022-07-28 at 11 46 10@2x" src="https://user-images.githubusercontent.com/5005153/181593157-7bb58a4f-ea8b-4e1d-b8e6-3f97f5e8247d.png">

**Reply by christoffee** [reply]
Yh it prints out snapshots count.... But screenshots are different... 

If you have 2 snapshots being tested on 2 screen widths and 2 browsers that would be 6 screenshots.

So I'm wondering if Percy can give you this screenshot count in the dryrun...

**Reply by wwilsman** [reply]
Unfortunately not. Since screenshots are based on project settings, like which configured browsers you have, we would need to query the API to find that information, which would require an API token and you to provide the `org/project` name for the settings endpoint. The dry-run feature intentionally doesn't make API requests as to not require a token, so there's no way to get that info while dry-running.

**Reply by christoffee** [reply]
ah that makes sense with the calculations of the screenshots including which browsers the user is testing on... but dont we have access to the screen widths per story ?

we could log out `10 snapshots, 30 screenshots x browsers supported` ?

**Reply by wwilsman** [reply]
Since individual snapshots can have their own widths, having a log like that would require us to keep track of snapshots and each of their widths to make an accurate calculation. Currently, the dry-run output just logs the size of the upload queue (a queue of functions), which isn't run while dry-running. So in summary, it would require a bit of a refactor to how the upload queue works in order to support this type of log. Which imo is incomplete anyway since we can't know the number of browsers.

If the queue happens to be refactored in the future to due other reasons, we might more easily be able to log something like this. But until then, it isn't really worth the effort of a refactor like that for a single log.
